From 72c4df81dbd1fc8fe31027f8a7497ff6b038129f Mon Sep 17 00:00:00 2001
From: "johan.bohlin" <johan.bohlin@intel.com>
Date: Wed, 24 Sep 2014 16:28:47 +0200
Subject: ASoC: Intel: Enable i2s_tristate_en pin on BYT-CR-RT5640

BT and Codec are both connected to SSP1, in order to get
bluetooth capture working i2s_tristate_en needs to be set.

Issue: GMIN-233

Change-Id: I0ab07713b1fb86ff04d2ce28699e2e792fddaaf7
Signed-off-by: johan.bohlin <johan.bohlin@intel.com>
---
 sound/soc/intel/board/byt_cr_dpcm_rt5640.c |   23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/sound/soc/intel/board/byt_cr_dpcm_rt5640.c b/sound/soc/intel/board/byt_cr_dpcm_rt5640.c
index 1917db3..e41ee3e 100644
--- a/sound/soc/intel/board/byt_cr_dpcm_rt5640.c
+++ b/sound/soc/intel/board/byt_cr_dpcm_rt5640.c
@@ -55,6 +55,7 @@ enum {
 	RT5640_GPIO_JD_INT2,
 	RT5640_GPIO_JACK_SWITCH,
 	RT5640_GPIO_JD_BUTTONS,
+	RT5640_GPIO_I2S_TRISTATE,
 };
 
 #define RT5640_GPIO_NA		-1
@@ -64,6 +65,7 @@ struct rt5640_gpios {
 	int jd_int2_gpio;
 	int jd_buttons_gpio;
 	int debug_mux_gpio;
+	int i2s_tristate_en_gpio;
 };
 
 struct byt_drvdata {
@@ -638,6 +640,27 @@ static int byt_init(struct snd_soc_pcm_runtime *runtime)
 		pr_warn("%s: GPIOs - JD-buttons: Not present!\n", __func__);
 	}
 
+	desc = devm_gpiod_get_index(codec->dev, NULL, RT5640_GPIO_I2S_TRISTATE);
+	if (!IS_ERR(desc)) {
+		drvdata->gpios.i2s_tristate_en_gpio = desc_to_gpio(desc);
+		devm_gpiod_put(codec->dev, desc);
+
+		byt_export_gpio(desc, "I2S-Tristate-En");
+		ret = gpiod_direction_output(desc, 0);
+		if (ret)
+			pr_warn("%s: Failed to set direction for GPIO%d (err = %d)!\n",
+				__func__, drvdata->gpios.i2s_tristate_en_gpio,
+				ret);
+
+		pr_info("%s: GPIOs - I2S-Tristate-En: %d (pol = %d, val = %d)\n",
+			__func__, drvdata->gpios.i2s_tristate_en_gpio,
+			gpiod_is_active_low(desc), gpiod_get_value(desc));
+	} else {
+		drvdata->gpios.i2s_tristate_en_gpio = RT5640_GPIO_NA;
+		pr_warn("%s: GPIOs - i2s_tristate_en-mux: Not present!\n",
+				__func__);
+	}
+
 	/* BYT-CR Audio Jack */
 
 	drvdata->t_jack_recheck = msecs_to_jiffies(BYT_T_JACK_RECHECK);
-- 
1.7.9.5

